### DISCUSSION TITLE
How to develop a complete website with Lit elements ?

### DISCUSSION THREAD

**Discussion by ghost** [lit/lit#3267] [Q&A] [answered]
I'm trying to build a complete website with Lit (vite+lit-ts+sass).

I want to create multiple web components inside the same repository, then import each custom element and use it to compose the pages, such as we do in a VDOM Framework....

I got some thinking about the project structure to do that, and I searched by some websites that already did the same, but without success...

And now I'm opening this question here in the community to try to learn more about this composition, if this is a good practice, if there is some example of a repository doing this on github...

Thanks in advance;

**Comment by ghost** [comment, marked as answer]
I found my solution and created a [repository](https://github.com/coheia/cohe-site) with the code...

**Reply by ttmc** [reply]
The linked repository no longer exists :(

**Reply by e111077** [reply]
FWIW I have a [sample project with vite+lit+ts+sass](https://github.com/e111077/vite-lit-element-ts-sass), but it only includes one component.

If you'd like to look at SPA structure to stack upon that example (vite+lit+ts+sass), personally I like [PWA Builder's approach](https://github.com/pwa-builder/pwa-starter). It also seems somewhat similar to the approach described.

There is also Open-WC 's CLI's app generation output. You can find the templates [here](https://github.com/open-wc/create/tree/master/src/generators/app-lit-element-ts/templates) instructions on how to use the CLI [here](https://open-wc.org/docs/development/generator/#development-generator).

The team does have an intention to eventually have a CLI generator for an app but that's not slated for anytime soon

**Reply by ghost** [reply]
> The linked repository no longer exists :(

@ttmc,  I'm sorry, I removed the private visibility...
